### ISSUE TITLE
TypeError: Bytes instead of String.... No plot

### ISSUE THREAD

**Issue by ludc** [facebookresearch/visdom#13]
Hi,

 I am trying to use Visdom with python 3.5. Each time I try to plot something, I obtain the following error (on, the server side ):

``` HTTPServerRequest(protocol='http', host='localhost:8097', method='POST', uri='/events', version='HTTP/1.1', remote_ip='127.0.0.1', headers={'Content-Length': '836', 'User-Agent': 'python-requests/2.12.4', 'Accept-Encoding': 'gzip, deflate', 'Host': 'localhost:8097', 'Accept': '*/*', 'Connection': 'keep-alive'})
Traceback (most recent call last):
  File "/home/denoyer/anaconda3/lib/python3.5/site-packages/tornado/web.py", line 1467, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "/home/denoyer/anaconda3/lib/python3.5/site-packages/visdom/server.py", line 227, in post
    args = json.loads(self.request.body)
  File "/home/denoyer/anaconda3/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'bytes'
ERROR:root:ERROR: 500: {'exc_info': (<class 'TypeError'>, TypeError("the JSON object must be str, not 'bytes'",), <traceback object at 0x7f7830e32888>)}
ERROR:tornado.access:500 POST /events (127.0.0.1) 2.10ms
```

Moreover, when plotting image, I have the error ```TypeError: string argument expected, got 'bytes'```on the client side -- I suppose it is a problem with the filename (line 479 in ```_save```, file ImageFile.py)

**Comment by elanmart** [comment]
@ludc Are you using `pip`? I've had similar error, seems it is fixed on `master` now.

**Comment by ludc** [comment]
Yes. I submitted a fix one hour ago.. I suppose it is related to my version of python...

**Comment by boyhero** [comment]
Has this been fixed? Should we reinstall without using pip?

**Comment by ludc** [comment]
I don't think so, but you can take my version until the merge github.com/ludc/visdom

**Comment by FlorianFranzen** [comment]
@ludc Thank you, your patch fixes the error for me:
```
pip install git+https://github.com/ludc/visdom.git
```

**Comment by ajabri** [comment]
Solved by #14.
